docs/ref/contrib/postgres/expressions.txt

```diff
@@ -0,0 +1,29 @@
+PostgreSQL specific query expressions
+=====================================
+
+.. currentmodule:: django.contrib.postgres.expressions
+
+``ArraySubquery``
+=================
+
+.. class:: ArraySubquery(queryset)
+
+``ArraySubquery`` is a ``Subquery`` subclass that uses the PostgreSQL ``ARRAY``
+constructor to build a list of values from the supplied queryset. Just like
+a regular subquery, it must only return a single column.
+
+This class differs from :class:`~django.contrib.postgres.aggregates.ArrayAgg`
+in the way that it does not act as an aggregate function and does not require
+a SQL ``GROUP BY`` clause to build the list of values.
+
+For example if one want to annotate all related books to an author as JSON
+objects::
+
+    >>> from django.db.models import OuterRef
+    >>> from django.db.models.functions import JSONObject
+    >>> from django.contrib.postgres.expressions import ArraySubquery
+    >>> books = Book.objects.filter(author=OuterRef('pk')).annotate(
+    ...     json=JSONObject(title="title", pages="pages")
+    ... ).values("json")
```

Comment:
I think you can skip the annotate call entirely

```suggestion
    >>> books = Book.objects.filter(author=OuterRef('pk')).values(
    ...     json=JSONObject(title="title", pages="pages")
    ... )
```

Reply:
Oh! TIL!